### ISSUE TITLE
Head mask set to None in the textVQA code

### ISSUE THREAD

**Issue by hanweiWill** [facebookresearch/pythia#230]
I noticed that the head mask input to the Bert model is set to None in M4C.py
`        head_mask = [None] * self.config.num_hidden_layers
        encoder_outputs = self.encoder(
            encoder_inputs,
            extended_attention_mask,
            head_mask=head_mask,
            relational_input = mmt_relational_input
        )`
then how do you make all decoding steps can only attend to previous decoding steps?

**Comment by ronghanghu** [comment]
Hi, the head mask only affects how many attention heads are used in self-attention (e.g. there are 12 heads by default, but can be pruned to fewer heads with `head_mask`). It has nothing to do with making a causal mask in time-steps.

The causal mask (a triangular mask that makes all decoding steps can only attend to previous decoding steps) is added here: https://github.com/facebookresearch/pythia/blob/project/m4c/pythia/models/m4c.py#L435-L436

**Comment by hanweiWill** [comment]
thank you for the explanation

**Comment by hanweiWill** [comment]
thank you for your explanation
